tcms/rpc/tests/test_testrun.py

```diff
@@ -106,6 +105,46 @@ def test_should_remove_a_case_run(self):
         self.assertFalse(TestExecution.objects.filter(pk=self.test_execution.pk).exists())
 
 
+class TestGetCases(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.test_case = TestCaseFactory()
+        self.test_case.save()
+        self.test_run = TestRunFactory()
+
+    def test_get_empty_result_if_no_case_added(self):
+        result = self.rpc_client.TestRun.get_cases(self.test_run.pk)
+        self.assertEqual(0, len(result))
+
+    def test_get_cases(self):
+        self.rpc_client.TestRun.add_case(self.test_run.pk, self.test_case.pk)
+        result = self.rpc_client.TestRun.get_cases(self.test_run.pk)
+        self.assertEqual(1, len(result))
+
+        case = result[0]
+        self.assertEqual(self.test_case.pk, case['id'])
+        self.assertTrue('execution_id' in case.keys())
+        self.assertTrue('status' in case.keys())
+
+
+class TestGetCasesPermission(APIPermissionsTestCase):
+    permission_label = 'testruns.view_testrun'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.test_run = TestRunFactory(plan=TestPlanFactory())
```

Comment:
You don't need to specify a plan in factory constructor unless you need a special test plan or you need to keep a reference to it for later use. Factories always create whatever objects they need and you don't have to specify them.